### PR TITLE
Fix #619 : Add integration test for UserJpaRepository

### LINKED ISSUE #619: UserJpaRepository: add missing integration test
Add Integration Tests for [UserJpaRepository](https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepository.java.mustache)

- clean a little bit the [unused variables](https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepository.java.mustache#L17-L19)
- add tests for each methods

### PR DESCRIPTION
**Pull request by beatfreaker**
Related to #619

### CODE REVIEW

**Comment by beatfreaker** [comment]
@pascalgrimaud test are failing because of `autoIncrement=true` is not present, should I fix it in this PR only, or it should be handle as separate issue ?

**Comment by pascalgrimaud** [comment]
Better to fix it separately, in another PR

**Comment by beatfreaker** [comment]
As we want to add `autoincrement` only for mysql is it okay to expect database in input request body as below
And use it in mustache template to render it conditionally ?

```json
{
  "folder": "/tmp/jhlite/undertow-simplecache",
  "generator-jhipster": {
    "projectName": "Beer Project",
    "baseName": "beer",
    "prettierDefaultIndent": 2,
    "packageName": "tech.jhipster.beer",
    “database”: “mysql”
  }
}

```

**Comment by pascalgrimaud** [comment]
No need, in liquibase code, we have the code which can know if it's MySQL database:
https://github.com/jhipster/jhipster-lite/blob/main/src/main/java/tech/jhipster/lite/generator/server/springboot/dbmigration/liquibase/domain/LiquibaseDomainService.java#L129

I can take care of this fix if you want?
I can address it in the next couple of minutes if needed

**Comment by pascalgrimaud** [comment]
- just done the fix https://github.com/jhipster/jhipster-lite/pull/640

Can you rebase plz?

File: src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepositoryIT.java.mustache [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  }
+
+  @Test
+  void shouldFindOneByActivationKey() {
+    UserEntity userEntity = userJpaRepository.findOneByActivationKey(ACTIVATION_KEY).orElse(null);
+    assertUserEntity(userEntity);
+  }
+
+  @Test
+  @Transactional
```
Comment (pascalgrimaud) [inline src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepositoryIT.java.mustache:45]: you can remove this `@Transactional`

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
Thanks and good job @beatfreaker !
This was important, as you can see, we manage to find some issues :)

**Comment by beatfreaker** [comment]
Bounty claimed at [https://opencollective.com/generator-jhipster/expenses/63101](https://opencollective.com/generator-jhipster/expenses/63101)

**Comment by pascalgrimaud** [comment]
@beatfreaker : approved !

**Pull request merged**